src/lib/jsonrpc.mli

```diff
@@ -39,7 +39,7 @@ val get : string -> (string * 'a) list -> 'a
 
 val call_of_string : string -> Rpc.call
 
-val version_id_and_call_of_string : string -> version * Rpc.t * Rpc.call
+val version_id_and_call_of_string : string -> version * (Rpc.t option) * Rpc.call
```

Comment:
This is the only real breaking change. I would like to take one more day to see if we can avoid it

Reply:
What I can do is let the old API as it was and create a new function suitable for my needs. Would it be okay for you?

Reply:
The current behaviour of the already existing function will be the same as it was (i.e throwing an exception if there's no id). This one will not support notifications, but I added a new function in order to support the absence of id in a request (making it a notification).

I think it'd be great to document the new functionality alongside with adding non-regression tests, as you suggested. May I do it in another PR, to have at least the current changes intergrated in master (if it's okay overall, and not making any breaking changes)? I can even add a part to the README on how to actually support notifications in the transport function, if you like.

Reply:
Yes please. Let’s add a notif_of_string function then. Would that work for you?

Reply:
Also the tests and documentation additions would be great to have. I am fine if you want to send them in a separate PR

Reply:
Earlier on I was thinking that maybe we could pass the call function to declare_ and then use call or notif accordingly without needing to change the type but I have not had time to think It through properly